network/Message.cpp

```diff
@@ -1309,3 +1318,16 @@ void ExtractAuthResponseMessageData(const Message& msg, std::string& player_name
 
 void ExtractSetAuthorizationRolesMessage(const Message &msg, Networking::AuthRoles& roles)
 { roles.SetText(msg.Text()); }
+
+void ExtractPlayerInfoMessageData(const Message &msg, std::map<int, PlayerInfo>& players) {
+    try {
+        std::istringstream is(msg.Text());
+        freeorion_xml_iarchive ia(is);
+        ia >> BOOST_SERIALIZATION_NVP(players);
+    } catch(const std::exception& err) {
+        ErrorLogger() << "ExtractPlayerInfo(const Message &msg, std::map<int, PlayerInfo>& players) failed!  Message:\n"
+                      << msg.Text() << "\n"
+                      << "Error: " << err.what();
+        throw err;
```

Comment:
Probably should rethrow original exception with ```throw;``` rather than throwing a sliced copy?

Reply:
Fixed